### PR TITLE
[BCF-3143] Multiple Address Bindings

### PR DESCRIPTION
**Pull request by EasterTheBunny**
### Description

Support for multiple address bindings in `ContractReader`.

### CODE REVIEW

File: pkg/solana/chainreader/chain_reader.go [hunk 1]
```diff
@@ -187,17 +205,36 @@ func (s *SolanaChainReaderService) QueryKey(_ context.Context, _ types.BoundCont
...
-	return s.bindings.Bind(bindings)
+	for _, binding := range bindings {
+		if err := s.bindings.Bind(binding); err != nil {
+			return err
+		}
+
+		s.lookup.bindAddressForContract(binding.Name, binding.Address)
+	}
+
+	return nil
```
Comment (ilija42) [inline pkg/solana/chainreader/chain_reader.go:216]: should Solana also have something like Binding registry to handle this in the same place>

File: pkg/solana/chainreader/chain_reader.go [hunk 2]
```diff
@@ -145,11 +163,11 @@ func (s *SolanaChainReaderService) GetLatestValue(ctx context.Context, readIdent
 			}
 
 			wg.Add(1)
-			go func(ctx context.Context, rb readBinding, res *loadedResult) {
+			go func(ctx context.Context, rb readBinding, res *loadedResult, address string) {
 				defer wg.Done()
 
-				rb.PreLoad(ctx, res)
-			}(localCtx, binding, results[idx])
+				rb.PreLoad(ctx, address, res)
```
Comment (ilija42) [inline pkg/solana/chainreader/chain_reader.go:169]: Is Preload similar to evm contract read? Why not call ReadAsync?

File: pkg/solana/chainreader/chain_reader.go (hunk 1 above)
Comment (EasterTheBunny) [reply pkg/solana/chainreader/chain_reader.go:216]: I think eventually, yes, but most of this will change when log events get rolled in.

**Review by ilija42 — APPROVED** [review]

File: pkg/solana/chainreader/chain_reader.go (hunk 2 above)
Comment (EasterTheBunny) [reply pkg/solana/chainreader/chain_reader.go:169]: For the reader, `PreLoad` does do an async read, but it stores the value in the reader and must be followed by `GetLatestValue`. It allows multiple readers to be read async, but in order.

**Review by aalu1418 — APPROVED** [review]

**Pull request merged**